flang/lib/Lower/Intrinsics.cpp

```diff
@@ -132,6 +140,7 @@ struct IntrinsicLibrary {
                                llvm::ArrayRef<mlir::Value> args);
 
   Fortran::lower::FirOpBuilder &builder;
+  mlir::Location loc;
```

Comment:
This utility was previously only using the location inside the FirOpBuilder, I feel like it should stick with it and get the location with `builder.getLoc()` when needed or get rid of all the createHere<> (*) otherwise the locations might go different ways here.

(*) technically, all the builder methods may also use its location under the hood (the character methods are), so I guess using `builder.getLoc()` in the IntrinsicLibrary ctor is the safest until the location is removed from the `FirOpBuilder`.

Reply:
I think we should get rid of createHere(). It is handy, but I fear it is error-prone -- see previous comment :).  The conversion helpers (like ConvertExpr, ConvertType, etc.) are meant to be very lightweight and cache this sort of information, while the bridge itself keeps the SPOT about the source position for the cursor point of what is being lowered "right now".